lib/gollum-lib/markups.rb

```diff
@@ -6,7 +6,7 @@ class Markup
         Kramdown::Document.new(content, :auto_ids => false, :input => "markdown").to_html
     }
 
-    register(:markdown,  "Markdown", :regexp => /md|mkdn?|mdown|markdown/)
+    register(:markdown,  "Markdown", :regexp => /md|mkdn?|mdown|markdown/, :h1 => /^(#\s+)(\w+.*)/)
```

Comment:
> is matching and mapping lines possible and reliable for all markups? I think header syntax for some markups could be multi-line...

I think multiline matches are doable. It's a bit of a hassle to contruct the right regexes, but we're doing currently doing something similar for code blocks.

> Maybe we should substitute the header content for direct character offset (start position of the header syntax) instead of line number?

By offset you mean [start-line number, end-line number]: good idea!

> `line.gsub(/^#{1}\K\s*.+$/, "GOLLUMSRCMP#{i}GOLLUMSRCMP")`

Thanks for the `\K` hint. This will obviously have to be a bit different if we're also going to need multiline matches, but I'll give it a shot!